apps/files_sharing/src/services/SharingService.ts

```diff
@@ -0,0 +1,181 @@
+/**
+ * @copyright Copyright (c) 2023 John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @author John Molakvoæ <skjnldsv@protonmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+/* eslint-disable camelcase, n/no-extraneous-import */
+import type { AxiosPromise } from 'axios'
+import type { ContentsWithRoot } from '../../../files/src/services/Navigation'
+
+import { Folder, File } from '@nextcloud/files'
+import { generateOcsUrl, generateRemoteUrl, generateUrl } from '@nextcloud/router'
+import { getCurrentUser } from '@nextcloud/auth'
+import axios from '@nextcloud/axios'
+import logger from './logger'
+
+export const rootPath = `/files/${getCurrentUser()?.uid}`
+
+export type OCSResponse = {
+	ocs: {
+		meta: {
+			status: string
+			statuscode: number
+			message: string
+		},
+		data: []
+	}
+}
+
+const headers = {
+	'Content-Type': 'application/json',
+}
+
+const ocsEntryToNode = function(ocsEntry: any): Folder | File | null {
```

Comment:
Any reason to not write it like so:

```suggestion
function ocsEntryToNode(ocsEntry: any): Folder | File | null {
```

?

Reply:
Personal preference and a few valid points:
- `const` prevents reassignment of the name while `function` does not
- same syntax as if it was an arrow function, while you cannot do that with `function`: `function () => {}`

In the end if looks as clean as any other variable, which in the end, it is.
And I like consistency ;)

Reply:
:+1: Indeed, personal preference, I was curious for the rational :)